src/components/order-service/order-service.controller.ts

```diff
@@ -94,7 +96,44 @@ export class OrderServiceController implements angular.IController {
     this.ctrl.selectedPlan = plan;
   }
 
+  public invalidNewProject() {
+    // test whether selectedProject is actually a newProject obj. and if passed name validation
+    return this.isNewProject() && !this.ctrl.selectedProject.validName;
+  }
+
   public provisionService() {
+    if (this.isNewProject()) {
+      // the selectedProject is actually a newProject object
+      let newProjName = this.ctrl.selectedProject.metadata.name;
+      let newProjDisplayName = this.ctrl.selectedProject.metadata.annotations['new-display-name'];
+      let newProjDesc = this.$filter('description')(this.ctrl.selectedProject);
+      let projReqObj: any = {
+        apiVersion: "v1",
+        kind: "ProjectRequest",
+        metadata: {
+          name: newProjName
+        },
+        displayName: newProjDisplayName,
+        description: newProjDesc
+      };
+      this.DataService
+        .create('projectrequests', null, projReqObj, this.$scope)
+        .then( (data: any) => {
+          this.createService();
+        }, (result: any) => {
+          var data = result.data || {};
+          if (data.reason === 'AlreadyExists') {
+            this.ctrl.nameTaken = true;
```

Comment:
Hey @dtaylor113 I was thinking since we already have the list of projects, can we check name taken before trying to create the project? We had a directive that does something like this in origin-web-console.

https://github.com/openshift/origin-web-console/blob/e7a0c0a8f703d5429f70b78223abb31856a66670/app/scripts/directives/oscUnique.js

Doesn't have to be in this PR